src/rules/percent-placeholder-pattern/index.js

```diff
@@ -0,0 +1,70 @@
+import _ from "lodash"
+import resolveNestedSelector from "postcss-resolve-nested-selector"
+import { utils } from "stylelint"
+import {
+  hasInterpolatingAmpersand,
+  isStandardRule,
+  isStandardSelector,
+  parseSelector,
+} from "../../utils"
+
+export const ruleName = "percent-placeholder-pattern"
+
+export const messages = utils.ruleMessages(ruleName, {
+  expected: placeholder => `Expected %-placeholder "%${placeholder}" to match specified pattern`,
+})
+
+export default function (pattern) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: pattern,
+      possible: [ _.isRegExp, _.isString ],
+    })
+    if (!validOptions) { return }
+
+    const placeholderPattern = (_.isString(pattern))
+      ? new RegExp(pattern)
+      : pattern
+
+    // Checking placeholder definitions (looking among regular rules)
+    root.walkRules(rule => {
+      const { selector } = rule
+
+      // If it's a custom prop or a less mixin
+      if (!isStandardRule(rule)) { return }
```

Comment:
The code below would rule out non-placeholder rules and declarations anyway, and this (and the following) checks are just early returns for not parsing the selector. Anyway, added a couple of tests for LESS.